### PR TITLE
feat(rust): add doc comments

### PR DESCRIPTION
**Pull request by clason**
replace old lua-match patterns with newly added nodes

### CODE REVIEW

File: queries/rust/highlights.scm
Comment (lucario387) [inline queries/rust/highlights.scm:224]: comment.documentation should only be captured for doc_comment I think

Comment (clason) [reply queries/rust/highlights.scm:224]: Hmm... Won't that mean you highlight the comment markers and the comments differently?

Comment (lucario387) [reply queries/rust/highlights.scm:224]: Yes.
I personally will do it like that. Unsure if that's the better behavior.

Comment (clason) [reply queries/rust/highlights.scm:224]: I have no skin in this game, so defer to the majority opinion. How do other languages handle that? (Consistency wins.)

Comment (clason) [reply queries/rust/highlights.scm:224]: I'm fine with either; I just want to hear the opinions before merging it to avoid later "this bad we need to change" PRs ;)

Comment (amaanq) [reply queries/rust/highlights.scm]: I think it should be the entire `line_comment`/`block_comment` node for consistency

Comment (ObserverOfTime) [reply queries/rust/highlights.scm]: I don't mind either way.

Comment (lucario387) [reply queries/rust/highlights.scm]: I'll go with @amaanq here then.

Comment (clason) [reply queries/rust/highlights.scm]: Then I'll go with the Rust maintainer's opinion. (This can always be revisited.)

**Review by ObserverOfTime — APPROVED** [review]

**Pull request merged**
